lib/adagrams.rb

```diff
@@ -0,0 +1,89 @@
+require "csv"
+
+# Helper method that creates array of letters that are available to be drawn
+def array_gen(hash)
+  array = []
+  hash.each do |key, value|
+    value.times do
+      array << key.to_s
+    end
+  end
+  return array
+end
+
+# Returns an array of 10 letters chosen at random from available letters
+def draw_letters
+  letter_freq = {
+    A: 9, N: 6, B: 2, O: 8, C: 2, P: 2, D: 4, Q: 1, E: 12, R: 6, F: 2, S: 4,
+    G: 3, T: 6, H: 2, U: 4, I: 9, V: 2, J: 1, W: 2, K: 1, X: 1, L: 4, Y: 2, M: 2, Z: 1,
+  }
+
+  avail_letters = array_gen(letter_freq)
+  used_letters = avail_letters.sample(10)
+
+  return used_letters
+end
+
+# Validates input word by determining if it can be formed with letters in hand
+def uses_available_letters?(input, letters_in_hand)
+  if !(input.is_a?(String))
+    raise ArgumentError, "Ummmmmm the value for the first argument needs to be a string, ok? Given value: #{input}"
+  elsif !(letters_in_hand.is_a?(Array))
+    raise ArgumentError, "The second argument should be an array. That doesn't look right...Given value: #{letters_in_hand}"
+  end
+  if input.length > letters_in_hand.length
+    return false
+  else
+    # reassigning letters_in_hand to new variable in order to avoid destruction of original array
+    possible_letters = letters_in_hand.clone
+    input.upcase.split(//).each do |char|
+      if possible_letters.include?(char)
+        possible_letters.delete(char)
```

Comment:
You will never return `true` for a word with a repeated letter. For example,
    ```ruby
    hand = ["E", "F", "R", "A", "E", "L", "V", "A", "D", "K"]
    uses_available_letters?("ADA", hand)
    # => false
    ```
    The reason why is that Ruby's `Array#delete` method will remove all matching elements from an array, not just the first match.

Reply:
Good to know! 
If we were to write something like:
`i = index_of (char)
possible_letters.delete_at(i), `
would it preserve the other repeats of the letter?
And is there a more elegant way of doing it?